### PR TITLE
Feature EPL-679: insert, update and delete operations was added and s…

### PR DESCRIPTION
**Pull request by giuliodoriEtendo**
…upported for DAS in DefaultFilter

### CODE REVIEW

**Review by juabaez — APPROVED** [review]

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 1]
```diff
@@ -35,40 +45,99 @@ public static String addFilters(String restMethod, String sql, String userId, St
...
+
+        if (containsWhere) {
+            return sql.replace(WHERE, WHERE + whereCondition + " AND ");
+        } else {
+            return sql + WHERE + whereCondition;
+        }
+    }
+
+
+    private static String replaceInQueryForDelete(String sql, String clientId, String orgId, boolean containsWhere, String tableAlias) {
```
Comment (Matias-Bernal) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:97]: Duplicate method

**Review by Matias-Bernal — APPROVED** [review]

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java [hunk 2]
```diff
@@ -35,40 +45,99 @@ public static String addFilters(String restMethod, String sql, String userId, St
...
         String tableAlias = getTableAlias(sql);
 
-        //RETURN DEFAULT FILTERS
-        return replaceInQuery(sql, clientId, orgId, isActive, containsWhere, tableAlias);
+        checkAccessByRole(sql, userId, clientId, orgId, roleId);
+
+        switch (restMethod) {
+            case GET:
+                return replaceInQueryForSelect(sql, clientId, orgId, isActive, containsWhere, tableAlias);
+            case POST:
```
Comment (szapata) [inline modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:57]: POST no sirve para Insertar y actualizar? con POST no estamos agregando filtros?

Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java:57]: POST solo para Insertar. PUT y PATCH para Actualizar (UPDATE).
Para los insert no se necesita filtros. Se deja un metodo DUMMY (checkAccessByRole) que sera implementado bajo otro SCOPE, donde se va a chequear los accesos a tablas y campos con el Role.

File: modules_core/com.etendorx.das/src/test/java/com/etendorx/das/utils/DefaultFiltersTest.java [hunk 3]
```diff
@@ -1,68 +1,203 @@
...
-        Assertions.assertEquals(sql, resultSql);
+    void testAddFilters_PostMethod() {
+        String userId = "user1";
+        String clientId = "client1";
+        String orgId = "org1";
+        String roleId = "role1";
+        String isActive = "true";
+        String restMethod = "POST";
+
+        String result = DefaultFilters.addFilters(select_QUERY, userId, clientId, orgId, roleId, isActive, restMethod);
```
[the remark this answers is not in the document]
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/test/java/com/etendorx/das/utils/DefaultFiltersTest.java:96]: What makes sense is that to insert, one or more SELECT type queries are performed to obtain the necessary data to be inserted.
As it is the POST method, it is validated that filters are not applied in those specific SELECTs

File: modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java (hunk 1 above)
Comment (giuliodoriEtendo) [reply modules_core/com.etendorx.das/src/main/java/com/etendorx/das/utils/DefaultFilters.java]: Done!

**Pull request merged**